infra/modules/azure_postgres_server/main.tf

```diff
@@ -0,0 +1,89 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.111.0"
+    }
+  }
+
+  required_version = "~> 1.7.5"
+}
+
+provider "azurerm" {
+  features {}
+
+  storage_use_azuread = true
+}
+
+module "naming_convention" {
+  source = "../azure_naming_convention"
+
+  environment = {
+    prefix          = var.environment.prefix
+    env_short       = var.environment.env_short
+    location        = var.environment.location
+    domain          = var.environment.domain
+    app_name        = var.environment.app_name
+    instance_number = var.environment.instance_number
+  }
+}
+
+#----------------------------#
+# PostgreSQL Flexible Server #
+#----------------------------#
+
+resource "azurerm_postgresql_flexible_server" "this" {
```

Comment:
should we enable autoscaling of storage? 

[`auto_grow_enabled = true`](https://registry.terraform.io/providers/hashicorp/azurerm/4.1.0/docs/resources/postgresql_flexible_server#auto_grow_enabled)

Reply:
Maybe we can enable this only for `premium` tier @Krusty93

Reply:
Perhaps for both `standard` and `premium` as both can be used in production?